src/adapters/items-fetcher.ts

```diff
@@ -0,0 +1,158 @@
+import LRU from 'lru-cache'
+import { AppComponents, Limits, ItemsResult, Item, ItemFetcher, ItemFetcherError, ItemFetcherErrorCode } from '../types'
+import { ISubgraphComponent } from '@well-known-components/thegraph-component'
+import { compareByRarity } from '../logic/utils'
+
+const THE_GRAPH_PAGE_SIZE = 1000
+
+// TODO cache metrics
+type ItemsQueryResponse = {
+  nfts: ItemFromQuery[]
+}
+
+type ItemFromQuery = {
+  urn: string
+  id: string
+  tokenId: string
+  transferredAt: number
+  item: {
+    rarity: string
+    price: number
+  }
+}
+
+const QUERY_EMOTES = createQueryForCategory('emote')
+const QUERY_WEARABLES = createQueryForCategory('wearable')
+
+function groupItemsByURN(items: ItemFromQuery[]): Item[] {
+  const itemsByURN = new Map<string, Item>()
+
+  items.forEach((item) => {
+    const individualData = {
+      id: item.id,
+      tokenId: item.tokenId,
+      transferredAt: item.transferredAt,
+      price: item.item.price
+    }
+    if (itemsByURN.has(item.urn)) {
+      const itemFromMap = itemsByURN.get(item.urn)!
+      itemFromMap.individualData.push(individualData)
+      itemFromMap.amount = itemFromMap.amount + 1
+    } else {
+      itemsByURN.set(item.urn, {
+        urn: item.urn,
+        individualData: [individualData],
+        rarity: item.item.rarity,
+        amount: 1
+      })
+    }
+  })
+
+  return Array.from(itemsByURN.values())
+}
+
+type ItemCategory = 'wearable' | 'emote'
+
+function createQueryForCategory(category: ItemCategory) {
+  return `query fetchItemsByOwner($owner: String, $idFrom: String) {
+    nfts(
+      where: { id_gt: $idFrom, owner: $owner, category: "${category}"},
+      orderBy: id,
+      orderDirection: asc,
+      first: ${THE_GRAPH_PAGE_SIZE}
+    ) {
+      urn,
+      id,
+      tokenId,
+      category
+      transferredAt,
+      item {
+        rarity,
+        price
+      }
+    }
+  }`
+}
+
+async function runItemsQuery(
+  subgraph: ISubgraphComponent,
+  address: string,
+  category: ItemCategory
+): Promise<ItemFromQuery[]> {
+  const items = []
+  const owner = address.toLowerCase()
+  let idFrom = ''
+  let result: ItemsQueryResponse
+  const query = category === 'wearable' ? QUERY_WEARABLES : QUERY_EMOTES
```

Comment:
What about using a kind of dictionary to avoid the conditional here and also to not use QUERY_EMOTES by default?, something like:

```
const queriesDictionary = {
  wearable: QUERY_WEARABLES,
  emote: QUERY_EMOTES
 }
```
So then:
`const query = queriesDictionary[category]`

Reply:
mmm not quite sure to follow, notice the type prevents for a third value that is not wearable or emote

Reply:
Yeah the `ItemCategory` being only two prevents that as Hugo says. But changed anyway because it looks nicer.